resources/aws_ec2.go

```diff
@@ -374,19 +383,28 @@ func (obj *AwsEc2Res) Watch() error {
 	if obj.WatchListenAddr != "" {
 		return obj.snsWatch()
 	}
-	return obj.longpollWatch()
+	return obj.waiterWatch()
 }
 
-// longpollWatch uses the ec2 api's built in methods to watch ec2 resource state.
-func (obj *AwsEc2Res) longpollWatch() error {
-	send := false
+// waiterWatch uses the ec2 api's built in methods to watch ec2 resource state.
+func (obj *AwsEc2Res) waiterWatch() error {
+	var send bool
 	var exit *error
+
+	// we tell the engine we're running right away. Because watchWaiter
```

Comment:
So the fact that we previously did this right away was wrong, and now we've got a comment saying it's right. It was wrong before, but it was a bug. Now the comment incorrectly re-enforces the bug. :/

obj.Running _tells_ the engine that we're now watching! This is a lie!! It's limited by their long poll API not exposing when they're ready, but still.

Reply:
I don't think I understand the potential consequences of this race. What exactly does this _mean_ to the engine and what could go wrong if we tell the engine we're watching a second or two before we poll the state and start the waiters?

Reply:
IOW, is this _just_ about potentially missing events, or is there more to this race on the engine side?